docs/01-app/05-api-reference/05-config/01-next-config-js/reactCompiler.mdx

```diff
@@ -4,9 +4,17 @@ description: Enable the React Compiler to automatically optimize component rende
 version: experimental
 ---
 
-Next.js 15 introduced support for the [React Compiler](https://react.dev/learn/react-compiler). The compiler improves performance by automatically optimizing component rendering. This reduces the amount of manual memoization developers have to do through APIs such as `useMemo` and `useCallback`.
+Next.js includes support for the [React Compiler](https://react.dev/learn/react-compiler), a tool designed to improve performance by automatically optimizing component rendering. This reduces the need for manual memoization using `useMemo` and `useCallback`.
 
-To use it, upgrade to Next.js 15, install the `babel-plugin-react-compiler`:
+Next.js includes a custom performance optimization written in SWC that makes the React Compiler more efficient. Instead of running the compiler on every file, Next.js analyzes your project and only applies the React Compiler to relevant files. This avoids unnecessary work and leads to faster builds compared to using the raw Babel plugin on its own.
```

Comment:
```suggestion
Next.js includes a custom performance optimization written in SWC that makes the React Compiler more efficient. Instead of running the compiler on every file, Next.js analyzes your project and only applies the React Compiler to relevant files. This avoids unnecessary work and leads to faster builds compared to using the Babel plugin on its own.
```

Reply:
This feels duplicated with the next paragraph under how it works, I would say we move this and combine there.